components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authcontext/JWTTokenGenerator.java

```diff
@@ -169,8 +169,7 @@ public void generateToken(OAuth2TokenValidationMessageContext messageContext) th
 
         RealmService realmService = OAuthComponentServiceHolder.getRealmService();
         // TODO : Need to handle situation where federated user name is similar to a one we have in our user store
-        if (realmService != null && tenantID != MultitenantConstants.INVALID_TENANT_ID &&
-                tenantID == OAuth2Util.getTenantIdFromUserName(authzUser)) {
```

Comment:
This == check is removed as with the below modification this check becomes a duplicate. Here the tenantID is always returning super tenant id as it was never set for AccessTokenDO class. Since  this is check is just to decide whether to check isExisting user this will be fine.